dashboard/15-final/app/lib/actions.ts

```diff
@@ -5,48 +5,86 @@ import { sql } from '@vercel/postgres';
 import { revalidatePath } from 'next/cache';
 import { redirect } from 'next/navigation';
 
-const InvoiceSchema = z.object({
+const FormSchema = z.object({
   id: z.string(),
-  customerId: z.string(),
-  amount: z.coerce.number(),
+  customerId: z.string({
+    invalid_type_error: 'Please select a customer.',
+  }),
+  amount: z.coerce
+    .number()
+    .gt(0, { message: 'Please enter an amount greater than $0.' }),
   status: z.enum(['pending', 'paid']),
   date: z.string(),
 });
-const CreateInvoice = InvoiceSchema.omit({ id: true, date: true });
-const UpdateInvoice = InvoiceSchema.omit({ date: true });
-const DeleteInvoice = InvoiceSchema.pick({ id: true });
 
-export async function createInvoice(formData: FormData) {
-  const { customerId, amount, status } = CreateInvoice.parse({
+const CreateInvoice = FormSchema.omit({ id: true, date: true });
+const UpdateInvoice = FormSchema.omit({ date: true });
+const DeleteInvoice = FormSchema.pick({ id: true });
+
+// This is temporary
```

Comment:
"temporary" how? Should it stay? Be remove?

Reply:
`useFormStatus` doesn't have types yet so we have to do this, I'm hoping by the time we launch this can be removed

Reply:
SGTM